Assignments/Program.cs

```diff
@@ -1 +1,110 @@
-﻿Console.WriteLine("Hello, World!");
+﻿// ---------------------------------------------------------------------------------------
+// Spark23 Assignments
+// Copyright (c) Metamation India.
+// ---------------------------------------------------------------------------------------
+// Program.cs
+// T2 - NUMBER TO WORDS AND ROMAN NUMERALS CONVERTER 
+// ---------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using static System.ConsoleColor;
+
+namespace Spark;
+
+public class T2 {
+   public static void Main () {
+      List<int> wTestCases = new () { 0, 1, 4, 9, 10, 11, 16, 25, 36, 49, 50, 81, 100, 109, 125, 216,
+         343, 729, 1000, 12001, 66000, 123450, 1000909, 987654321, 100000001 };
+      List<int> rTestCases = new () { 1, 4, 5, 9, 10, 47, 94, 100, 345, 400, 499, 500, 900, 998,
+         1000, 1500, 1729, 1888, 3495, 3913, 4000, 4096, 4509, 4761, 4999 };
+      Console.WriteLine ("\x1B[4m" + "Numbers to Words:" + "\x1B[0m");
+      for (int i = 0; i < wTestCases.Count; i++)
+         PrintResult (i, wTestCases[i], true);
+      Console.WriteLine ("\n\x1B[4m" + "Numbers to Romans:" + "\x1B[0m");
+      for (int i = 0; i < rTestCases.Count; i++)
+         PrintResult (i, rTestCases[i], false);
+   }
+
+   /// <summary>Prints the words or roman numerals equivalent to the input number</summary>
+   public static void PrintResult (int index, int input, bool toWords) {
+      Console.Write ($"{index + 1,2}) {input,10} ==> ");
+      Console.ForegroundColor = Yellow;
+      Console.WriteLine (toWords ? ToWords (input) : ToRomans (input));
+      Console.ResetColor ();
+   }
+
+   /// <summary>Retuns the input number in words using State Machine</summary>
+   public static string ToWords (int num) {
+      if (num == 0) return "Zero";
+      string input = num.ToString ().PadLeft (9, '0'),
+            output = "";
+      List<string> ones = new () { "", "One", "Two", "Three", "Four", "Five", "Six", "Seven", "Eight", "Nine", "Ten",
+      "Eleven", "Twelve", "Thirteen", "Fourteen", "Fifteen", "Sixteen", "Seventeen", "Eighteen", "Nineteen" },
+                   tens = new () { "", "Ten", "Twenty", "Thirty", "Forty", "Fifty", "Sixty", "Seventy", "Eighty", "Ninety" };
+
+      // As this methods works till 9-digit number, this first loop segregates the input (string) into 3 + 3 + 3 digits, and parse into integer.
+      for (int i = 0; i < 3; i++) {
+         // Here part is FIRST 3-digits for (i = 0), SECOND 3-digits for (i = 1), and LAST 3-digits for (i = 2).

```

Comment:
You can aggregate all the comments for the entire logic in a paragraph above the first loop.